### PR TITLE
Fixing TransformerXL bool issue #1169

### PR DESCRIPTION
**Pull request by thomwolf**
Fixing #1169 regarding using uint or bool masks in Transformer-XL and PyTorch 1.1.0 and 1.2.0.

Hopefully, this solution will be compatible upward with the future PyTorch releases.

### CODE REVIEW

File: pytorch_transformers/modeling_transfo_xl.py [hunk 1]
```diff
@@ -423,7 +423,8 @@ def forward(self, h, attn_mask=None, mems=None, head_mask=None):
         # [qlen x klen x bsz x n_head]
         attn_score = torch.einsum('ibnd,jbnd->ijbn', (head_q, head_k))
         attn_score.mul_(self.scale)
-        if attn_mask is not None and attn_mask.any().item():
+        if attn_mask is not None and torch.sum(attn_mask).item():
+            attn_mask = (attn_mask == 1)  # Switch to bool
```
Comment (LysandreJik) [inline pytorch_transformers/modeling_transfo_xl.py]: That's very clean

**Comment by LysandreJik** [comment]
Looks great to me!
